### PR TITLE
Handle simultaneous channel restarts

### LINKED ISSUE #6022: Simultaneous timeouts in JobHost worker cause "Number of initialized language workers exceeded"
Two functions can time out at the same time and cause the same worker to crash for independent reasons. If this channel is a webhost channel, we handle that correctly. However, if this is a jobhost-scoped channel, we do not handle for this condition:
https://github.com/Azure/azure-functions-host/blob/2d0ac22afb86041a5b81797b6aaf16049a673198/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs#L321

Per Yong:
1. App is NodeJS with HTTP triggered functions (could be any app).
2. Customer has two functions that trigger timeout exception (30 seconds) almost at the same time.
3. The same channel running those functions is restarted & disposed almost at same time.
4. Two node.exe process are started almost at the same time as well while listening on the same tcp port 52088, which is unexpected.
5. This leads to the final exception Number of initialized language workers exceeded:2 exceeded maxProcessCount: 1 while executing the function:

Note that any worker can get into this state after at least one fatal worker exception on the instance.

cc: @pragnagopa and @yojagad

### PR DESCRIPTION
**Pull request by yojagad**
Fixes #6022

### CODE REVIEW

File: src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs [hunk 1]
```diff
@@ -308,30 +308,17 @@ public async void WorkerRestart(WorkerRestartEvent workerRestart)
...
+            _logger.LogDebug("Attempting to dispose webhost or jobhost channel for workerId: {channelId}, runtime:{language}", workerId, runtime);
 
-            // Dispose old worker
-            if (isWebHostChannel)
-            {
-                _logger.LogDebug("Disposing WebHost channel for workerId: {channelId}, for runtime:{language}", workerId, runtime);
-            }
-            else
+            bool isWebHostChannelDisposed = await _webHostLanguageWorkerChannelManager.ShutdownChannelIfExistsAsync(runtime, workerId);
+            bool isJobHostChannelDisposed = await _jobHostLanguageWorkerChannelManager.ShutdownChannelIfExistsAsync(workerId);
```
Comment (mhoeger) [inline src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs]: although it's super super small, I think the "TryRemove" in the concurrent dictionary isn't totally trivial: https://stackoverflow.com/questions/15252115/concurrentdictionary-performance-at-a-single-thread-misunderstanding

Maybe worth checking if we need to do "isJobHostChannelDisposed" only if "!isWebHostChannelDisposed" but it's also hard to say without really benchmarking it and it's probably not that important to..!

Comment (pragnagopa) [reply src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs]: +1 check !isWebHostChannelDisposed before looking at jobHostChannels

**Review by pragnagopa — DISMISSED** [review]
just a minor comment. Rest looks good.

**Review by pragnagopa — APPROVED** [review]
👍

**Pull request merged**